joel@Joels-MacBook-Pro scripts % ./ivy-tendril-dev -b tendril/00518-FixActionBarWrapRegressionAndImplementProgressiveButtonColla
=== Fetching and switching to 'tendril/00518-FixActionBarWrapRegressionAndImplementProgressiveButtonColla' in /Users/joel/ivy/Ivy-Tendril ===
branch 'tendril/00518-FixActionBarWrapRegressionAndImplementProgressiveButtonColla' set up to track 'origin/tendril/00518-FixActionBarWrapRegressionAndImplementProgressiveButtonColla'.
Switched to a new branch 'tendril/00518-FixActionBarWrapRegressionAndImplementProgressiveButtonColla'
=== Starting Ivy.Tendril (dev repo) ===
/Users/joel/ivy/Ivy-Tendril/src/Ivy.Tendril/Helpers/ActionBarResponsive.cs(41,16): error CS0029: Cannot implicitly convert type 'Ivy.DropDownMenu' to 'Ivy.Button'
/Users/joel/ivy/Ivy-Tendril/src/Ivy.Tendril/Helpers/ActionBarResponsive.cs(50,16): error CS0029: Cannot implicitly convert type 'Ivy.DropDownMenu' to 'Ivy.Button'

The build failed. Fix the build errors and run again.

--- a/src/Ivy.Tendril/Apps/Drafts/ActionBarView.cs
+++ b/src/Ivy.Tendril/Apps/Drafts/ActionBarView.cs
@@ -111,100 +111,90 @@ public class ActionBarView(
             })
         };
 
-        // Desktop dropdown items: buttons that don't fit at Desktop tier + standard overflow
-        var desktopDropdownItems = new List<MenuItem>
+        // Split / Expand share the same OnClick logic whether shown as an inline button
+        // (Full tier) or a dropdown item (Compact/Minimal tiers).
+        void StartSplit()
+        {
+            if (hasActiveSplitJob) return;
+            var optimisticPlan = selectedPlan with
+            {
+                Metadata = selectedPlan.Metadata with { State = PlanStatus.Updating }
+            };
+            selectedPlanState.Set(optimisticPlan);
+            planService.TransitionState(selectedPlan.FolderName, PlanStatus.Updating);
+            jobService.StartJob(new SplitPlanArgs(selectedPlan.FolderPath));
+            refreshPlans();
+        }
+
+        void StartExpand()
+        {
+            if (hasActiveExpandJob) return;
+            var optimisticPlan = selectedPlan with
+            {
+                Metadata = selectedPlan.Metadata with { State = PlanStatus.Building }
+            };
+            selectedPlanState.Set(optimisticPlan);
+            planService.TransitionState(selectedPlan.FolderName, PlanStatus.Building);
+            jobService.StartJob(new ExpandPlanArgs(selectedPlan.FolderPath));
+            refreshPlans();
+        }
+
+        // Compact-tier dropdown items: buttons that don't fit at the Compact tier + standard overflow
+        var compactDropdownItems = new List<MenuItem>
         {
             new MenuItem("Split", Icon: Icons.Scissors, Tag: "Split")
-                .OnSelect(() =>
-                {
-                    if (hasActiveSplitJob) return;
-                    var optimisticPlan = selectedPlan with
-                    {
-                        Metadata = selectedPlan.Metadata with { State = PlanStatus.Updating }
-                    };
-                    selectedPlanState.Set(optimisticPlan);
-                    planService.TransitionState(selectedPlan.FolderName, PlanStatus.Updating);
-                    jobService.StartJob(new SplitPlanArgs(selectedPlan.FolderPath));
-                    refreshPlans();
-                })
-                .Disabled(hasActiveSplitJob),
+                .OnSelect(StartSplit).Disabled(hasActiveSplitJob),
             new MenuItem("Expand", Icon: Icons.UnfoldVertical, Tag: "Expand")
-                .OnSelect(() =>
-                {
-                    if (hasActiveExpandJob) return;
-                    var optimisticPlan = selectedPlan with
-                    {
-                        Metadata = selectedPlan.Metadata with { State = PlanStatus.Building }
-                    };
-                    selectedPlanState.Set(optimisticPlan);
-                    planService.TransitionState(selectedPlan.FolderName, PlanStatus.Building);
-                    var planPath = selectedPlan.FolderPath;
-                    jobService.StartJob(new ExpandPlanArgs(planPath));
-                    refreshPlans();
-                })
-                .Disabled(hasActiveExpandJob),
+                .OnSelect(StartExpand).Disabled(hasActiveExpandJob),
             new MenuItem("Delete", Icon: Icons.Trash, Tag: "Delete").OnSelect(showDeleteDialog)
         };
-        desktopDropdownItems.AddRange(standardOverflowItems);
+        compactDropdownItems.AddRange(standardOverflowItems);
 
-        // Mobile dropdown items: all action buttons + standard overflow
-        var mobileDropdownItems = new List<MenuItem>
+        // Minimal-tier dropdown items: all action buttons + standard overflow
+        var minimalDropdownItems = new List<MenuItem>
         {
             new MenuItem("Edit", Icon: Icons.Pencil, Tag: "Edit")
                 .OnSelect(() => isEditingState.Set(true)),
             new MenuItem("Update", Icon: Icons.WandSparkles, Tag: "Update")
                 .OnSelect(showUpdateDialog),
             new MenuItem("Split", Icon: Icons.Scissors, Tag: "Split")
-                .OnSelect(() =>
-                {
-                    if (hasActiveSplitJob) return;
-                    var optimisticPlan = selectedPlan with
-                    {
-                        Metadata = selectedPlan.Metadata with { State = PlanStatus.Updating }
-                    };
-                    selectedPlanState.Set(optimisticPlan);
-                    planService.TransitionState(selectedPlan.FolderName, PlanStatus.Updating);
-                    jobService.StartJob(new SplitPlanArgs(selectedPlan.FolderPath));
-                    refreshPlans();
-                })
-                .Disabled(hasActiveSplitJob),
+                .OnSelect(StartSplit).Disabled(hasActiveSplitJob),
             new MenuItem("Expand", Icon: Icons.UnfoldVertical, Tag: "Expand")
-                .OnSelect(() =>
-                {
-                    if (hasActiveExpandJob) return;
-                    var optimisticPlan = selectedPlan with
-                    {
-                        Metadata = selectedPlan.Metadata with { State = PlanStatus.Building }
-                    };
-                    selectedPlanState.Set(optimisticPlan);
-                    planService.TransitionState(selectedPlan.FolderName, PlanStatus.Building);
-                    var planPath = selectedPlan.FolderPath;
-                    jobService.StartJob(new ExpandPlanArgs(planPath));
-                    refreshPlans();
-                })
-                .Disabled(hasActiveExpandJob),
+                .OnSelect(StartExpand).Disabled(hasActiveExpandJob),
             new MenuItem("Delete", Icon: Icons.Trash, Tag: "Delete").OnSelect(showDeleteDialog)
         };
-        mobileDropdownItems.AddRange(standardOverflowItems);
+        minimalDropdownItems.AddRange(standardOverflowItems);
 
-        // Action bar without .Wrap() - single row layout with progressive collapse
+        // Action bar without .Wrap() - single row layout with progressive collapse.
+        // Full tier (>=1024px): all buttons inline + overflow-only dropdown.
+        // Compact tier (768-1023px): Previous, Next, Edit, Update inline; Split/Expand/Delete in dropdown.
+        // Minimal tier (<768px): Previous, Next inline; everything else in dropdown.
         return Layout.Horizontal().AlignContent(Align.Left).Gap(2)
-               // Desktop: Previous, Next, Edit, Update visible
                | new Button("Previous").Icon(Icons.ChevronLeft).Outline().OnClick(goToPrevious)
                    .ShortcutKey("p").AlwaysVisible()
                | new Button("Next").Icon(Icons.ChevronRight, Align.Right).Outline().OnClick(goToNext)
                    .ShortcutKey("n").AlwaysVisible()
                | new Button("Edit").Icon(Icons.Pencil).Outline().ShortcutKey("E")
-                   .OnClick(() => isEditingState.Set(true)).DesktopUp()
+                   .OnClick(() => isEditingState.Set(true)).CompactUp()
                | new Button("Update").Icon(Icons.WandSparkles).Outline().ShortcutKey("u")
-                   .OnClick(showUpdateDialog).DesktopUp()
-               // Desktop dropdown: Split, Expand, Delete + standard overflow
-               | ActionBarResponsive.DropdownAtDesktop(
+                   .OnClick(showUpdateDialog).CompactUp()
+               | new Button("Split").Icon(Icons.Scissors).Outline()
+                   .OnClick(StartSplit).Disabled(hasActiveSplitJob).FullOnly()
+               | new Button("Expand").Icon(Icons.UnfoldVertical).Outline()
+                   .OnClick(StartExpand).Disabled(hasActiveExpandJob).FullOnly()
+               | new Button("Delete").Icon(Icons.Trash).Outline()
+                   .OnClick(showDeleteDialog).FullOnly()
+               // Full-tier dropdown: standard overflow items only
+               | ActionBarResponsive.DropdownAtFull(
                    new Button().Icon(Icons.EllipsisVertical).Ghost(),
-                   desktopDropdownItems.ToArray())
-               // Mobile dropdown: all action buttons + standard overflow
-               | ActionBarResponsive.DropdownAtMobile(
+                   standardOverflowItems)
+               // Compact-tier dropdown: Split, Expand, Delete + standard overflow
+               | ActionBarResponsive.DropdownAtCompact(
                    new Button().Icon(Icons.EllipsisVertical).Ghost(),
-                   mobileDropdownItems.ToArray());
+                   compactDropdownItems.ToArray())
+               // Minimal-tier dropdown: all action buttons + standard overflow
+               | ActionBarResponsive.DropdownAtMinimal(
+                   new Button().Icon(Icons.EllipsisVertical).Ghost(),
+                   minimalDropdownItems.ToArray());
     }
 }

--- a/src/Ivy.Tendril/Apps/Recommendations/ContentView.cs
+++ b/src/Ivy.Tendril/Apps/Recommendations/ContentView.cs
@@ -175,46 +175,54 @@ public class ContentView(
             })
         };
 
-        // Desktop dropdown: View Plan + standard overflow
-        var desktopDropdownItems = new List<MenuItem>
+        void ViewPlan()
         {
-            new MenuItem("View Plan", Icon: Icons.ExternalLink, Tag: "ViewPlan").OnSelect(() =>
-            {
-                var fullPath = Path.Combine(planService.PlansDirectory, selectedRecommendation.PlanFolderName);
-                if (Directory.Exists(fullPath))
-                    showPlan(fullPath);
-            })
-        };
-        desktopDropdownItems.AddRange(standardOverflowItems);
+            var fullPath = Path.Combine(planService.PlansDirectory, selectedRecommendation.PlanFolderName);
+            if (Directory.Exists(fullPath))
+                showPlan(fullPath);
+        }
 
-        // Mobile dropdown: Accept with Notes, View Plan + standard overflow
-        var mobileDropdownItems = new List<MenuItem>
+        // Full-tier dropdown: standard overflow items only (all buttons shown inline)
+        var fullDropdownItems = standardOverflowItems;
+
+        // Compact-tier dropdown: View Plan + standard overflow
+        var compactDropdownItems = new List<MenuItem>
+        {
+            new MenuItem("View Plan", Icon: Icons.ExternalLink, Tag: "ViewPlan").OnSelect(ViewPlan)
+        };
+        compactDropdownItems.AddRange(standardOverflowItems);
+
+        // Minimal-tier dropdown: Accept with Notes, View Plan + standard overflow
+        var minimalDropdownItems = new List<MenuItem>
         {
             new MenuItem("Accept with Notes", Icon: Icons.CircleCheck, Tag: "AcceptWithNotes")
                 .OnSelect(() => showNotesDialog()),
-            new MenuItem("View Plan", Icon: Icons.ExternalLink, Tag: "ViewPlan").OnSelect(() =>
-            {
-                var fullPath = Path.Combine(planService.PlansDirectory, selectedRecommendation.PlanFolderName);
-                if (Directory.Exists(fullPath))
-                    showPlan(fullPath);
-            })
+            new MenuItem("View Plan", Icon: Icons.ExternalLink, Tag: "ViewPlan").OnSelect(ViewPlan)
         };
-        mobileDropdownItems.AddRange(standardOverflowItems);
+        minimalDropdownItems.AddRange(standardOverflowItems);
 
-        // Action bar without .Wrap() - single row with progressive collapse
+        // Action bar without .Wrap() - single row with progressive collapse.
+        // Full (>=1024px): Previous, Next, Accept with Notes, View Plan inline + overflow dropdown.
+        // Compact (768-1023px): Previous, Next, Accept with Notes inline; View Plan in dropdown.
+        // Minimal (<768px): Previous, Next inline; everything else in dropdown.
         var actionBar = Layout.Horizontal().AlignContent(Align.Left).Gap(2)
                         | new Button("Previous").Icon(Icons.ChevronLeft).Outline().ShortcutKey("p")
                             .OnClick(GoToPrevious).AlwaysVisible()
                         | new Button("Next").Icon(Icons.ChevronRight, Align.Right).Outline().ShortcutKey("n")
                             .OnClick(GoToNext).AlwaysVisible()
                         | new Button("Accept with Notes").Icon(Icons.CircleCheck).Outline().ShortcutKey("w")
-                            .OnClick(() => showNotesDialog()).DesktopUp()
-                        | ActionBarResponsive.DropdownAtDesktop(
+                            .OnClick(() => showNotesDialog()).CompactUp()
+                        | new Button("View Plan").Icon(Icons.ExternalLink).Outline()
+                            .OnClick(ViewPlan).FullOnly()
+                        | ActionBarResponsive.DropdownAtFull(
                             new Button().Icon(Icons.EllipsisVertical).Ghost(),
-                            desktopDropdownItems.ToArray())
-                        | ActionBarResponsive.DropdownAtMobile(
+                            fullDropdownItems)
+                        | ActionBarResponsive.DropdownAtCompact(
                             new Button().Icon(Icons.EllipsisVertical).Ghost(),
-                            mobileDropdownItems.ToArray());
+                            compactDropdownItems.ToArray())
+                        | ActionBarResponsive.DropdownAtMinimal(
+                            new Button().Icon(Icons.EllipsisVertical).Ghost(),
+                            minimalDropdownItems.ToArray());
 
         var mainLayout = new HeaderLayout(
             header,

--- a/src/Ivy.Tendril/Apps/Review/ContentView.cs
+++ b/src/Ivy.Tendril/Apps/Review/ContentView.cs
@@ -383,38 +383,49 @@ public class ContentView(
             })
         };
 
-        // Desktop dropdown: Discard + standard overflow
-        var desktopDropdownItems = new List<MenuItem>
+        // Full-tier dropdown: standard overflow items only (all buttons shown inline)
+        var fullDropdownItems = standardOverflowItems;
+
+        // Compact-tier dropdown: Discard + standard overflow
+        var compactDropdownItems = new List<MenuItem>
         {
             new MenuItem("Discard", Icon: Icons.Trash, Tag: "Discard").OnSelect(showDiscardDialog)
         };
-        desktopDropdownItems.AddRange(standardOverflowItems);
+        compactDropdownItems.AddRange(standardOverflowItems);
 
-        // Mobile dropdown: all action buttons + standard overflow
-        var mobileDropdownItems = new List<MenuItem>
+        // Minimal-tier dropdown: all action buttons + standard overflow
+        var minimalDropdownItems = new List<MenuItem>
         {
             new MenuItem("Reset to Draft", Icon: Icons.RotateCcw, Tag: "ResetToDraft").OnSelect(showResetToDraftDialog),
             new MenuItem("Request Changes", Icon: Icons.MessageSquare, Tag: "RequestChanges").OnSelect(showSuggestChangesDialog),
             new MenuItem("Discard", Icon: Icons.Trash, Tag: "Discard").OnSelect(showDiscardDialog)
         };
-        mobileDropdownItems.AddRange(standardOverflowItems);
+        minimalDropdownItems.AddRange(standardOverflowItems);
 
-        // Action bar without .Wrap() - single row with progressive collapse
+        // Action bar without .Wrap() - single row with progressive collapse.
+        // Full (>=1024px): Previous, Next, Reset to Draft, Request Changes, Discard inline + overflow dropdown.
+        // Compact (768-1023px): Previous, Next, Reset to Draft, Request Changes inline; Discard in dropdown.
+        // Minimal (<768px): Previous, Next inline; everything else in dropdown.
         return Layout.Horizontal().AlignContent(Align.Left).Gap(2)
                 | new Button("Previous").Icon(Icons.ChevronLeft).Outline().OnClick(() => GoToPrevious(nav, args))
                     .ShortcutKey("p").AlwaysVisible()
                 | new Button("Next").Icon(Icons.ChevronRight, Align.Right).Outline().OnClick(() => GoToNext(nav, args))
                     .ShortcutKey("n").AlwaysVisible()
                 | new Button("Reset to Draft").Icon(Icons.RotateCcw).Outline().ShortcutKey("r")
-                    .OnClick(showResetToDraftDialog).DesktopUp()
+                    .OnClick(showResetToDraftDialog).CompactUp()
                 | new Button("Request Changes").Icon(Icons.MessageSquare).Outline().ShortcutKey("c")
-                    .OnClick(showSuggestChangesDialog).DesktopUp()
-                | ActionBarResponsive.DropdownAtDesktop(
+                    .OnClick(showSuggestChangesDialog).CompactUp()
+                | new Button("Discard").Icon(Icons.Trash).Outline()
+                    .OnClick(showDiscardDialog).FullOnly()
+                | ActionBarResponsive.DropdownAtFull(
                     new Button().Icon(Icons.EllipsisVertical).Ghost(),
-                    desktopDropdownItems.ToArray())
-                | ActionBarResponsive.DropdownAtMobile(
+                    fullDropdownItems)
+                | ActionBarResponsive.DropdownAtCompact(
                     new Button().Icon(Icons.EllipsisVertical).Ghost(),
-                    mobileDropdownItems.ToArray());
+                    compactDropdownItems.ToArray())
+                | ActionBarResponsive.DropdownAtMinimal(
+                    new Button().Icon(Icons.EllipsisVertical).Ghost(),
+                    minimalDropdownItems.ToArray());
     }
 
     private object BuildContent(

--- a/src/Ivy.Tendril/Helpers/ActionBarResponsive.cs
+++ b/src/Ivy.Tendril/Helpers/ActionBarResponsive.cs
@@ -3,20 +3,29 @@ namespace Ivy.Tendril.Helpers;
 /// <summary>
 /// Helper for creating responsive action bars with progressive button collapse.
 /// Buttons collapse into dropdown menus as screen size decreases.
+///
+/// IMPORTANT — breakpoint bands (see <see cref="Breakpoint"/> and the frontend
+/// <c>use-responsive.ts</c>): a breakpoint name maps to a width *band*, not a
+/// "this width and up" range:
+/// <list type="bullet">
+///   <item><c>Mobile</c>  → width &lt; 640px</item>
+///   <item><c>Tablet</c>  → 640px ≤ width &lt; 768px</item>
+///   <item><c>Desktop</c> → 768px ≤ width &lt; 1024px</item>
+///   <item><c>Wide</c>    → width ≥ 1024px</item>
+/// </list>
+/// A normal/large desktop monitor (≥1024px) therefore resolves to <c>Wide</c>, NOT
+/// <c>Desktop</c>. The three collapse tiers used by the action bars map to bands as:
+/// <list type="bullet">
+///   <item><b>Full</b> tier (everything visible) → <c>Wide</c></item>
+///   <item><b>Compact</b> tier (some buttons collapsed) → <c>Desktop</c></item>
+///   <item><b>Minimal</b> tier (most buttons collapsed) → <c>Mobile</c> + <c>Tablet</c></item>
+/// </list>
 /// </summary>
 public static class ActionBarResponsive
 {
     /// <summary>
-    /// Shows button only on Desktop breakpoint and above (>=768px).
-    /// Hidden on Mobile and Tablet.
-    /// </summary>
-    public static Button DesktopUp(this Button btn)
-    {
-        return btn.ShowOn(Breakpoint.Desktop);
-    }
-
-    /// <summary>
-    /// Shows button at all breakpoints (always visible).
+    /// Shows button at all breakpoints (always visible). Used for navigation
+    /// (Previous/Next) which must stay visible at every size.
     /// </summary>
     public static Button AlwaysVisible(this Button btn)
     {
@@ -24,28 +33,48 @@ public static class ActionBarResponsive
     }
 
     /// <summary>
-    /// Shows button only below Desktop breakpoint (&lt;768px).
-    /// Visible on Mobile and Tablet only.
+    /// Shows button only at the Full tier — a wide desktop monitor (width ≥1024px).
+    /// Hidden at the Compact and Minimal tiers, where the button collapses into a dropdown.
     /// </summary>
-    public static Button BelowDesktop(this Button btn)
+    public static Button FullOnly(this Button btn)
     {
-        return btn.ShowOn(Breakpoint.Mobile, Breakpoint.Tablet);
+        return btn.ShowOn(Breakpoint.Wide);
     }
 
     /// <summary>
-    /// Creates a dropdown menu that is visible only on Desktop and above (>=768px).
-    /// Hidden on Mobile and Tablet.
+    /// Shows button at the Compact tier and up — a wide monitor or the 768–1023px band
+    /// (<c>Desktop</c> + <c>Wide</c>). Hidden only at the Minimal tier (&lt;768px), where
+    /// the button collapses into a dropdown.
     /// </summary>
-    public static DropDownMenu DropdownAtDesktop(Button trigger, params MenuItem[] items)
+    public static Button CompactUp(this Button btn)
+    {
+        return btn.ShowOn(Breakpoint.Desktop, Breakpoint.Wide);
+    }
+
+    /// <summary>
+    /// Creates a dropdown menu visible only at the Full tier (width ≥1024px). Holds the
+    /// standard overflow items when all action buttons are shown inline.
+    /// </summary>
+    public static DropDownMenu DropdownAtFull(Button trigger, params MenuItem[] items)
+    {
+        return trigger.WithDropDown(items).ShowOn(Breakpoint.Wide);
+    }
+
+    /// <summary>
+    /// Creates a dropdown menu visible only at the Compact tier (768–1023px band).
+    /// Holds the buttons collapsed at this tier plus the standard overflow items.
+    /// </summary>
+    public static DropDownMenu DropdownAtCompact(Button trigger, params MenuItem[] items)
     {
         return trigger.WithDropDown(items).ShowOn(Breakpoint.Desktop);
     }
 
     /// <summary>
-    /// Creates a dropdown menu that is visible only below Desktop (&lt;768px).
-    /// Visible on Mobile and Tablet only.
+    /// Creates a dropdown menu visible only at the Minimal tier (width &lt;768px,
+    /// i.e. <c>Mobile</c> + <c>Tablet</c>). Holds the buttons collapsed at this tier
+    /// plus the standard overflow items.
     /// </summary>
-    public static DropDownMenu DropdownAtMobile(Button trigger, params MenuItem[] items)
+    public static DropDownMenu DropdownAtMinimal(Button trigger, params MenuItem[] items)
     {
         return trigger.WithDropDown(items).ShowOn(Breakpoint.Mobile, Breakpoint.Tablet);
     }

--- a/src/Ivy.Tendril/Helpers/ActionBarResponsive.cs
+++ b/src/Ivy.Tendril/Helpers/ActionBarResponsive.cs
@@ -36,7 +36,7 @@ public static class ActionBarResponsive
     /// Creates a dropdown menu that is visible only on Desktop and above (>=768px).
     /// Hidden on Mobile and Tablet.
     /// </summary>
-    public static Button DropdownAtDesktop(Button trigger, params MenuItem[] items)
+    public static DropDownMenu DropdownAtDesktop(Button trigger, params MenuItem[] items)
     {
         return trigger.WithDropDown(items).ShowOn(Breakpoint.Desktop);
     }
@@ -45,7 +45,7 @@ public static class ActionBarResponsive
     /// Creates a dropdown menu that is visible only below Desktop (&lt;768px).
     /// Visible on Mobile and Tablet only.
     /// </summary>
-    public static Button DropdownAtMobile(Button trigger, params MenuItem[] items)
+    public static DropDownMenu DropdownAtMobile(Button trigger, params MenuItem[] items)
     {
         return trigger.WithDropDown(items).ShowOn(Breakpoint.Mobile, Breakpoint.Tablet);
     }

--- a/src/Ivy.Tendril/Ivy.Tendril.csproj
+++ b/src/Ivy.Tendril/Ivy.Tendril.csproj
@@ -44,10 +44,10 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.Data.Sqlite" Version="10.0.0" />
     <PackageReference Include="Microsoft.Extensions.AI.OpenAI" Version="10.4.1" />
-    <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions" Version="10.0.5" />
-    <PackageReference Include="Microsoft.Extensions.Configuration.UserSecrets" Version="10.0.5" />
-    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="10.0.5" />
-    <PackageReference Include="Microsoft.Extensions.Http" Version="10.0.5" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions" Version="10.0.9" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.UserSecrets" Version="10.0.9" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="10.0.9" />
+    <PackageReference Include="Microsoft.Extensions.Http" Version="10.0.9" />
     <PackageReference Include="ModelContextProtocol" Version="1.2.0" />
     <PackageReference Include="OpenAI" Version="2.9.1" />
     <PackageReference Include="PostHog" Version="2.6.0" />

